cadquery/cq.py

```diff
@@ -1046,6 +1051,43 @@ def compounds(
         """
         return self._selectObjects("Compounds", selector, tag)
 
+    def ancestors(self: T, kind: Shapes, tag: Optional[str] = None) -> T:
+        """
+        Iterate over ancestors.
+
+        :param kind: kind of ancestor, e.g. "Face" or "Edge"
+        :param tag: if set, search the tagged object instead of self
+        :return: a Workplane object who's stack contains all ancestors.
+
+
+        """
+        ctx_solid = self.findSolid()
+        objects = self._getTagged(tag).objects if tag else self.objects
+
+        results = [
+            el.ancestors(ctx_solid, kind) for el in objects if isinstance(el, Shape)
+        ]
+
+        return self.newObject(set(el for res in results for el in res))
+
+    def siblings(self: T, kind: Shapes, level: int = 1, tag: Optional[str] = None) -> T:
+        """
+        Iterate over siblings.
+
+        :param kind: kind of linking element, e.g. "Vertex" or "Edge"
+        :param level: level of relation - how many elements of kind are in the link
+        :param tag: if set, search the tagged object instead of self
+        :return: a Workplane object who's stack contains all ancestors.
```

Comment:
change "ancestors" to "siblings".

Change the return value docstring to something like?:
```        :return: a Workplane object whose stack contains the selected siblings.```

or ```... contains the siblings.```

Reply:
Fixed